### PR TITLE
Automagically configure mingw on Windows

### PR DESCRIPTION
**Pull request by pmuetschard**
- No more need for `--config mingw` on Windows
 - No more absolute paths:
   - uses %CC% if given or looks for gcc in the %PATH%
   - uses gcc to figure out the system include paths

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -6,32 +6,39 @@ Bazel is able to fetch most of the dependencies required to build GAPID, but cur
...
+
+---
+
 ## Windows
 
 ### Install Chocolatey
 
 [Follow these instructions](https://chocolatey.org/install) to install Chocolatey.
 
-### Install Java Runtime 8
```
Comment (ben-clayton) [inline BUILDING.md]: Why did you remove this?
It may not be needed for bazel, but is unfortunately needed to do anything with the Android SDK. 😦

File: BUILDING.md [hunk 2]
```diff
@@ -40,9 +47,7 @@ To fetch the required packages, using a console type:
 
 ```
 cd <sdk-path>
-tools\bin\sdkmanager "platforms;android-21"
-tools\bin\sdkmanager "platforms;android-27"
-tools\bin\sdkmanager "build-tools;26.0.1
+tools\bin\sdkmanager platforms;android-21 build-tools;26.0.1
```
Comment (ben-clayton) [inline BUILDING.md]: Have you tested without android 27?

File: BUILDING.md [hunk 3]
```diff
@@ -55,43 +60,32 @@ Either do this globally or in your shell every time.
...
-cd <path-to-gapid-source>
-bazel build --config mingw pkg
-```
-The build output will be at `bazel-bin/pkg`.
+| Variable            | Target                             |
+| ------------------- | ---------------------------------- |
+| `ANDROID_HOME`      | Path to Android SDK                |
+| `ANDROID_NDK_HOME`  | Path to Android NDK                |
+| `BAZEL_SH`          | `C:\tools\msys64\usr\bin\bash.exe` |
+| `TMP` and `TEMP`    | `C:\tmp`                           |
```
Comment (ben-clayton) [inline BUILDING.md]: `TMP` seems to work without also changing `TEMP`.

File: BUILDING.md [hunk 4]
```diff
@@ -55,43 +60,32 @@ Either do this globally or in your shell every time.
...
 
 The following environment variables will need to be set prior to building:
 
-| Variable            | Target                            |
-| ------------------- | --------------------------------- |
-| `JAVA_HOME`         | Path to the Java Runtime          |
-| `ANDROID_HOME`      | Path to Android SDK               |
-| `ANDROID_NDK_HOME`  | Path to Android NDK               |
-| `BAZEL_SH`          | `C:\tools\msys64\usr\bin\bash.exe`|
-| `GOPATH`            | \<unset\>                         |
```
Comment (ben-clayton) [inline BUILDING.md]: So `GOPATH` and `GOROOT` are no longer needed to be unset?

File: BUILDING.md (hunk 2 above)
Comment (pmuetschard) [reply BUILDING.md]: Yes. I don't have android 27 installed, and the macOS kokoro build doesn't install it either.

File: BUILDING.md (hunk 4 above)
Comment (pmuetschard) [reply BUILDING.md]: No, I tested it and it doesn't seem to have an impact. I know I was the source of this, sorry about that.

File: BUILDING.md (hunk 3 above)
Comment (pmuetschard) [reply BUILDING.md]: Done.

File: BUILDING.md (hunk 1 above)
Comment (pmuetschard) [reply BUILDING.md]: Done.
